src/vcf/assembly_checker.cpp

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright 2014-2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "vcf/assembly_checker.hpp"
+
+namespace ebi
+{
+  namespace vcf
+  {
+    namespace assembly_checker
+    {
+
+      bool check_vcf_ref(std::istream &vcf_input,
+                         std::istream &fasta_input,
+                         std::istream &fasta_index_input,
+                         std::vector<std::unique_ptr<ebi::vcf::AssemblyReportWriter>> &outputs)
+      {
+          std::vector<char> vector_line;
+          vector_line.reserve(default_line_buffer_size);
+
+          // Reading FASTA index, and querying FASTA file
+          auto index = bioio::read_fasta_index(fasta_index_input);
+
+          size_t line_num = 0;
+          while (util::readline(vcf_input, vector_line).size() != 0) {
+              std::string line{vector_line.begin(), vector_line.end()};
+
+              if (boost::starts_with(line, "#")) {
+                  continue;
+              }
+
+              RecordCore record_core = build_record_core(line,++line_num);
```

Comment:
`line_num` should not be incremented here because it won't count the header lines. But you could replace the while with a for loop:

```c++
for (size_t line_num = 1; util::readline(vcf_input, vector_line).size() != 0; line_num++) {
    // ...
    RecordCore record_core = build_record_core(line, line_num);
    // ...
}
```